static/app/utils/profiling/flamegraph/flamegraphThemeProvider.tsx

```diff
@@ -10,22 +11,47 @@ import {
 
 export const FlamegraphThemeContext = createContext<FlamegraphTheme | null>(null);
 
+type FlamegraphThemeMutationCallback = (
```

Comment:
I assume you need this because we need to mutate this from the children component right?
It smell to me like we should stop passing entire theme object to the renderers and only pass the values required by each renderer - this way we get some flexibility back and stop relying on the theme.

Reply:
Absolutely we should be passing via props.

In an ideal scenario some of our flamegraph ui components shouldn't rely so much on context. But then again, we have to weigh in on how often we're going to use a flamegraph elsewhere.  🤷🏼

Reply:
>Absolutely we should be passing via props.

We did this at the start and it was just a lot of prop drilling hence why we removed it, but at least for each individual renderer, we should only be passing what we need. I have not foreseen a case where we'd want to make this dynamic (not from like a child component pov like here), but hindsight is 20/20 or something - we should keep this in mind as we are imo approaching a time when we might want to improve this part of the codebase